lib/counter.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const assert = require('assert');
+const { EventEmitter } = require('events');
+const Metric = require('@metrics/metric');
+
+const Counter = class Counter extends EventEmitter {
+    constructor(options) {
+        super();
+
+        assert(
+            typeof options === 'object',
+            `argument 'options' must be provided`,
+        );
+
+        assert(
+            /^[A-Za-z-_]*$/.test(options.name),
+            `argument 'options.name' must be a valid string`,
+        );
+
+        assert(
+            options.description && typeof options.description === 'string',
+            `argument 'options.description' must be provided`,
+        );
+
+        this.options = {
+            labels: [],
+            ...options,
+        };
+    }
+
+    inc(value, ...rest) {
+        let val = value;
+
+        if (typeof value !== 'number') {
+            val = 1;
+
+            if (typeof value === 'string') {
+                rest.unshift(value);
+            }
+        }
+
+        const labels = this.options.labels.map((label, i) => ({
+            name: label,
+            value: rest[i],
+        }));
+
+        const metric = new Metric({
+            name: this.options.name,
+            description: this.options.description,
+            timestamp: Date.now() / 1000,
```

Comment:
`timestamp` can be dropped. The new metric object will do this as the default value.